lib/pre2017.js

```diff
@@ -0,0 +1,97 @@
+const fetch = require('node-fetch')
+const cheerio = require('cheerio')
+const https = require('https')
+
+const PRE_URL = [
+  'https://codein.withgoogle.com/archive/2016',
+  'https://codein.withgoogle.com/archive/2015',
+  'https://www.google-melange.com/archive/gci/2014',
+  'https://www.google-melange.com/archive/gci/2013',
+  'https://www.google-melange.com/archive/gci/2012',
+  'https://www.google-melange.com/archive/gci/2011',
+  'https://www.google-melange.com/archive/gci/2010',
+]
+
+async function fetchText(url) {
+  const agent = new https.Agent({
+    rejectUnauthorized: false,
+  })
+  const res = await fetch(url, {
+    agent,
+  })
+  return await res.text()
+}
+
+async function fetchPre2017(url) {
+  const year = url.split('/').pop()
+  const responseHTML = await fetchText(url)
```

Comment:
Fetching and scraping process should be separated.

Export the scraping function and test it.